src/amp-react-utils.js

```diff
@@ -53,9 +53,10 @@ export function useStateFromProp(prop) {
 /**
  * @param {!Element} elementRef
  * @param {function(number, number)} callback
+ * @param {!Array=} opt_deps
  */
-export function useResizeEffect(elementRef, callback) {
-  useMountEffect(() => {
+export function useResizeEffect(elementRef, callback, opt_deps) {
+  useEffect(() => {
```

Comment:
Why was this changed to an effect? Won't this register mutation observers?

Reply:
This won't register mutation observers, but it's definitely somewhat more wasteful. However, I was looking for the least error-prone situation. The real change here: we initially implemented `useResizeEffect` via `useEffect(..., [/*empty array == mount effect*/])`. With this change, I know allow the last argument (deps array) to be passed by the caller. I also not entirely convinced this is a good thing, but here's what happens. 

There are two reasons why we may need to recalculate font size in this component:
 1. The layout size of the component changes.
 2. The component's `props` change. Not all `props` changes require recalc, but it's not always easy to tell which would. Definitely `min/maxFontSize` do. And with `props.children` it's simply impossible to tell at all.

The old implementation only catches element resizing itself. And I do feel like it's "true" to what it promises. The big benefit here is that it resolves the element's width and height completely asynchronously - there's no blocking reads there at all. However, this effect is NOT rerun when the component's children/props change. I.e. we only support (1) from above. To note, however, async width/height resolution is great for a generic `useResizeEffect`, but we are not truly able to take advantage of it in this particular case because we still have to do numerous blocking calls to other DOM APIs to calculate the optimal font size.

The new implementation forces uses `useResizeEffect` on each rerender, so it naturally supports (1) and (2). We still get the component's width/height in a non-blocking fashion (but again, impact is muted as described above). This is a bit more wasteful, because on every rerender it unsubscribes from the `ResizeObserver` and then resubscribes again. Not sure how bad it is in practice.

The big question I have is whether this is contrary to what's expected from `useResizeObserver`. Optimally the side-effect should only fire when an element changes. If anything the correct `deps` value should logically be `[elementRef.current]`. If we wanted to keep `useResizeObserver` as it is now, the code would change to:

```

function AmpFitText(props) {
  const containerRef = useRef();

  ...
  function update() {
    // font-size calculations.
  }

  // Recalc every time size changes.
  useResizeEffect(containerRef, update);
  
  // Recalc every time props change.
  useEffect(update);

  return (/*render*/);
}
```